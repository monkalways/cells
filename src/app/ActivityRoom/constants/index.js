// DMU Service URLS
const GET_ACTIVITY_ROOM_DETAINEES_URL = (usage) => `${
  process.env.REACT_APP_CELL_SERVICE_URL
}api/cells/${usage}/get-activity-room-detainees`;

const GET_LAST_TEMP_ABSENCE_URL = (detaineeId) => `${
  process.env.REACT_APP_CELL_SERVICE_URL
}api/detainees/${detaineeId}/last-temporary-absence`;

const CREATE_TEMP_ABSENCE_URL = `${
  process.env.REACT_APP_CELL_SERVICE_URL
}api/temp-absences/`;

const UPDATE_TEMP_ABSENCE_URL = `${
  process.env.REACT_APP_CELL_SERVICE_URL
}api/temp-absences/`;

export default {
  GET_ACTIVITY_ROOM_DETAINEES_URL,
  GET_LAST_TEMP_ABSENCE_URL,
  CREATE_TEMP_ABSENCE_URL,
  UPDATE_TEMP_ABSENCE_URL,
};
