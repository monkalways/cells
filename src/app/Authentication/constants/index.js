const AUTHENTICATE_URL = `${
  process.env.REACT_APP_CELL_SERVICE_URL
}api/current-user`;

export default {
  AUTHENTICATE_URL,
};
