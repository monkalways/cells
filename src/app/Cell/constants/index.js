const CELL_DETAILS_URL = 'api/cells/';

const HEADERS = {
  headers: {
    Accept: 'application/json,',
    'x-work-domain': 'edmonton',
    authorization: 'Basic Y3BiMTpkZXZlMzMzMw==',
  },
};

export default {
  CELL_DETAILS_URL,
  HEADERS,
};
