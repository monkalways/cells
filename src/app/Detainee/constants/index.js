// URLs
const AVAILABLE_ACTIVITY_ROOMS = 'get-activity-rooms-availability';
const CELLS_URL = 'api/cells/';
const DETAINEE_URL = 'api/detainees/';
const DETAINEE_PROFILE = '/detainee-profile';
const CELLS_QUERYSTRING = 'cells';

// Activity Rooms
const BAIL_HEARING_ROOM_1 = 'Bail Hearing1';
const BAIL_HEARING_ROOM_2 = 'Bail Hearing2';
const BREATH_TEST_ROOM = 'Breath Test';
const FINGERPRINTING_ROOM = 'Fingerprinting';
const INTERVIEW_ROOM = 'Interview';
const MEDICAL_ROOM = 'Medical';
const PHONE_ROOM = 'Phone';

// Temporary Absence Reasons
const BAIL_HEARING_IN_PROGRESS = 'Bail Hearing - In Progress';
const BAIL_HEARING_IN_TRANSIT = 'Bail Hearing - In Transit';
const BREATH_TEST_IN_PROGRESS = 'Breath Test - In Progress';
const BREATH_TEST_IN_TRANSIT = 'Breath Test - In Transit';
const FINGERPRINTING_IN_PROGRESS = 'Fingerprinting - In Progress';
const FINGERPRINTING_IN_TRANSIT = 'Fingerprinting - In Transit';
const INTERVIEW_IN_PROGRESS = 'Interview - In Progress';
const INTERVIEW_IN_TRANSIT = 'Interview - In Transit';
const MEDICAL_IN_PROGRESS = 'Medical - In Progress';
const MEDICAL_IN_TRANSIT = 'Medical - In Transit';
const PHONE_IN_PROGRESS = 'Phone - In Progress';
const PHONE_IN_TRANSIT = 'Phone - In Transit';

export default {
  AVAILABLE_ACTIVITY_ROOMS,
  BAIL_HEARING_ROOM_1,
  BAIL_HEARING_ROOM_2,
  BAIL_HEARING_IN_PROGRESS,
  BAIL_HEARING_IN_TRANSIT,
  BREATH_TEST_IN_PROGRESS,
  BREATH_TEST_IN_TRANSIT,
  BREATH_TEST_ROOM,
  CELLS_QUERYSTRING,
  CELLS_URL,
  DETAINEE_URL,
  DETAINEE_PROFILE,
  FINGERPRINTING_IN_PROGRESS,
  FINGERPRINTING_IN_TRANSIT,
  FINGERPRINTING_ROOM,
  INTERVIEW_ROOM,
  INTERVIEW_IN_PROGRESS,
  INTERVIEW_IN_TRANSIT,
  MEDICAL_ROOM,
  MEDICAL_IN_PROGRESS,
  MEDICAL_IN_TRANSIT,
  PHONE_ROOM,
  PHONE_IN_PROGRESS,
  PHONE_IN_TRANSIT,
};
