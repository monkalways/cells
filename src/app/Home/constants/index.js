const VERSION_URL = 'api/version';

export default {
  VERSION_URL,
};
