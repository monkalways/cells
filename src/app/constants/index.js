export const WELFARE_ACTION_ACCEPT = 'Accept';
export const WELFARE_ACTION_REJECT = 'Reject';
export const WELFARE_ACTION_NOTAPPLICABLE = 'NotApplicable';
export const WELFARE_ACTION_VISUAL = 'Visual';
export const WELFARE_ACTION_VERBAL = 'Verbal';

export const LOG_TYPE_DETAINEE_ACTION = 'DetaineeAction';
export const LOG_TYPE_WELFARE_ISMEAL = 'IsMeal';
export const LOG_TYPE_WELFARE_ISMEDICATION = 'IsMedication';
export const LOG_TYPE_WELFARE_ISCELLCHECK = 'IsCellCheck';
export const DETAINEE_ACTION_PHONE_DECLINED = 'PhoneDeclined';

export const PHONE_IN_TRANSIT = 'Phone - In Transit';
export const PHONE_IN_PROGRESS = 'Phone - In Progress';
export const MEDICAL_IN_TRANSIT = 'Medical - In Transit';
export const MEDICAL_IN_PROGRESS = 'Medical - In Progress';
export const INTERVIEW_IN_TRANSIT = 'Interview - In Transit';
export const INTERVIEW_IN_PROGRESS = 'Interview - In Progress';
export const FINGERPRINTING_IN_TRANSIT = 'Fingerprinting - In Transit';
export const FINGERPRINTING_IN_PROGRESS = 'Fingerprinting - In Progress';
export const BREATH_TEST_IN_TRANSIT = 'Breath Test - In Transit';
export const BREATH_TEST_IN_PROGRESS = 'Breath Test - In Progress';
export const BAIL_HEARING_IN_TRANSIT = 'Bail Hearing - In Transit';
export const BAIL_HEARING_IN_PROGRESS = 'Bail Hearing - In Progress';
export const REMAND_HOLDING_IN_TRANSIT = 'Remand Holding - In Transit';
export const RELEASE_HOLDING_IN_TRANSIT = 'Release Holding - In Transit';
export const CELL_IN_TRANSIT = 'Cell - In Transit';
