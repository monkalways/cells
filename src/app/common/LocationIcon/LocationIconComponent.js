import React from 'react';
import PropTypes from 'prop-types';
import { Badge, withStyles } from '@material-ui/core';
import constants from '../../constants';

import CellIcon from '../../images/InCell.png';
import InTransitIcon from '../../images/InTransit.png';
import BailHearingIcon from '../../images/BailHearingGeneral.png';
import BreathTestIcon from '../../images/Breathalyzer.png';
import FingerprintingIcon from '../../images/Fingerprinting.png';
import InterviewIcon from '../../images/Interview.png';
import MedicalIcon from '../../images/MedicalVisit.png';
import PhoneIcon from '../../images/PhoneAccept.png';
import ReleaseIcon from '../../images/ReleaseHolding.png';
import RemandIcon from '../../images/RemandHolding.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  location: PropTypes.string.isRequired,
};

export const LocationIconComponent = ({ classes, location }) => {
  const getLocationIcon = () => {
    if (location) {
      if (location.includes(constants.IN_TRANSIT)) {
        return InTransitIcon;
      }

      switch (location) {
        case constants.BAIL_HEARING_IN_PROGRESS:
          return BailHearingIcon;
        case constants.BREATH_TEST_IN_PROGRESS:
          return BreathTestIcon;
        case constants.FINGERPRINTING_IN_PROGRESS:
          return FingerprintingIcon;
        case constants.INTERVIEW_IN_PROGRESS:
          return InterviewIcon;
        case constants.MEDICAL_IN_PROGRESS:
          return MedicalIcon;
        case constants.PHONE_IN_PROGRESS:
          return PhoneIcon;
        default:
          return CellIcon;
      }
    }
    return CellIcon;
  };

  const getTransitDestinationIcon = () => {
    switch (location) {
      case constants.BREATH_TEST_IN_TRANSIT:
        return BreathTestIcon;
      case constants.BAIL_HEARING_IN_TRANSIT:
        return BailHearingIcon;
      case constants.CELL_IN_TRANSIT:
        return CellIcon;
      case constants.FINGERPRINTING_IN_TRANSIT:
        return FingerprintingIcon;
      case constants.INTERVIEW_IN_TRANSIT:
        return InterviewIcon;
      case constants.MEDICAL_IN_TRANSIT:
        return MedicalIcon;
      case constants.PHONE_IN_TRANSIT:
        return PhoneIcon;
      case constants.RELEASE_HOLDING_IN_TRANSIT:
        return ReleaseIcon;
      case constants.REMAND_HOLDING_IN_TRANSIT:
        return RemandIcon;
      default:
        return null;
    }
  };

  const isDetaineeInTransit = () => {
    if (location) {
      return location.includes(constants.IN_TRANSIT);
    }
    return false;
  };

  return (
    <Badge
      classes={{ badge: classes.badge }}
      badgeContent={
        isDetaineeInTransit() ? (
          <img
            className={classes.cornerImg}
            alt="Transit Destination"
            src={getTransitDestinationIcon()}
          />
        ) : (
          <div />
        )
      }
    >
      <img
        className={classes.mainImg}
        alt="Detainee Location"
        src={getLocationIcon()}
      />
    </Badge>
  );
};

LocationIconComponent.propTypes = propTypes;

export default withStyles(() => ({
  badge: {
    height: '41.6%',
    width: '41.6%',
    top: '58.3%',
    right: 0,
  },
  cornerImg: {
    height: '100%',
    width: '100%',
  },
  mainImg: {
    height: '100%',
    width: '100%',
  },
}))(LocationIconComponent);
