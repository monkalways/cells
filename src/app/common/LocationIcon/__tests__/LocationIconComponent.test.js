import React from 'react';
import { shallow } from 'enzyme';
import constants from '../../../constants';

import { LocationIconComponent } from '../LocationIconComponent';

describe('LocationIconComponent', () => {
  let location;

  beforeEach(() => {
    location = '';
  });

  const setup = () => {
    const classes = {
      badge: 'badge',
      cornerImg: 'cornerImg',
      mainImg: 'mainImg',
    };

    return shallow(<LocationIconComponent classes={classes} location={location} />);
  };

  describe('Location Icon', () => {
    it('should render In Transit icon', () => {
      location = `Some place - ${constants.IN_TRANSIT}`;
      const wrapper = setup();
      expect(wrapper.find('img')).toHaveProp('src', 'InTransit.png');
    });
    it('should render Bail Hearing icon', () => {
      location = constants.BAIL_HEARING_IN_PROGRESS;
      const wrapper = setup();
      expect(wrapper.find('img')).toHaveProp('src', 'BailHearingGeneral.png');
    });
    it('should render Breath Test icon', () => {
      location = constants.BREATH_TEST_IN_PROGRESS;
      const wrapper = setup();
      expect(wrapper.find('img')).toHaveProp('src', 'Breathalyzer.png');
    });
    it('should render Fingerprinting icon', () => {
      location = constants.FINGERPRINTING_IN_PROGRESS;
      const wrapper = setup();
      expect(wrapper.find('img')).toHaveProp('src', 'Fingerprinting.png');
    });
    it('should render Interview icon', () => {
      location = constants.INTERVIEW_IN_PROGRESS;
      const wrapper = setup();
      expect(wrapper.find('img')).toHaveProp('src', 'Interview.png');
    });
    it('should render Medical icon', () => {
      location = constants.MEDICAL_IN_PROGRESS;
      const wrapper = setup();
      expect(wrapper.find('img')).toHaveProp('src', 'MedicalVisit.png');
    });
    it('should render Phone icon', () => {
      location = constants.PHONE_IN_PROGRESS;
      const wrapper = setup();
      expect(wrapper.find('img')).toHaveProp('src', 'PhoneAccept.png');
    });
    it('should render Cell icon', () => {
      location = '';
      const wrapper = setup();
      expect(wrapper.find('img')).toHaveProp('src', 'InCell.png');
    });
  });

  describe('Transit Destination Icon', () => {
    it('should render Breath Test icon', () => {
      location = constants.BREATH_TEST_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render Bail Hearing icon', () => {
      location = constants.BAIL_HEARING_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render Cell icon', () => {
      location = constants.CELL_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render Fingerprinting icon', () => {
      location = constants.FINGERPRINTING_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render Interview icon', () => {
      location = constants.INTERVIEW_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render Medical icon', () => {
      location = constants.MEDICAL_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render Phone icon', () => {
      location = constants.PHONE_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render Release Holding icon', () => {
      location = constants.RELEASE_HOLDING_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render Remand Holding icon', () => {
      location = constants.REMAND_HOLDING_IN_TRANSIT;
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
