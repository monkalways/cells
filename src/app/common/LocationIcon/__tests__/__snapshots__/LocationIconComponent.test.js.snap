// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationIconComponent Transit Destination Icon should render Bail Hearing icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="BailHearingGeneral.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;

exports[`LocationIconComponent Transit Destination Icon should render Breath Test icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="Breathalyzer.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;

exports[`LocationIconComponent Transit Destination Icon should render Cell icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="InCell.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;

exports[`LocationIconComponent Transit Destination Icon should render Fingerprinting icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="Fingerprinting.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;

exports[`LocationIconComponent Transit Destination Icon should render Interview icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="Interview.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;

exports[`LocationIconComponent Transit Destination Icon should render Medical icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="MedicalVisit.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;

exports[`LocationIconComponent Transit Destination Icon should render Phone icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="PhoneAccept.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;

exports[`LocationIconComponent Transit Destination Icon should render Release Holding icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="ReleaseHolding.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;

exports[`LocationIconComponent Transit Destination Icon should render Remand Holding icon 1`] = `
<WithStyles(Badge)
  badgeContent={
    <img
      alt="Transit Destination"
      className="cornerImg"
      src="RemandHolding.png"
    />
  }
  classes={
    Object {
      "badge": "badge",
    }
  }
>
  <img
    alt="Detainee Location"
    className="mainImg"
    src="InTransit.png"
  />
</WithStyles(Badge)>
`;
